### PR TITLE
add cabal support for some libs

### PR DESCRIPTION
**Pull request by kpinter-da**
This PR:
* adds and updates cabal files for some packages so they may be referenced as a dependency
* splits up daml-lf-proto to multiple smaller libraries in order to avoid a circular dependency that Bazel can handle but Cabal can't
* deletes 2 invalid symlinks that cause stack to fail when referencing this repo as an extra dependency

### CODE REVIEW

File: compiler/damlc/stable-packages/BUILD.bazel [hunk 1]
```diff
@@ -32,7 +32,7 @@ da_haskell_binary(
         "optparse-applicative",
         "text",
     ],
-    main_function = "GenerateStablePackage.main",
+    main_function = "Main.main",
```
Comment (akrmn) [inline compiler/damlc/stable-packages/BUILD.bazel:35]:
```suggestion
```
since this is the default, I'd just remove the argument

**Comment by dylant-da** [comment]
@kpinter-da Where are you using this? We're looking at programmatically generating the cabal file using bazel, so that we can keep the bazel file as the source of truth about dependencies.

**Comment by kpinter-da** [comment]
@dylant-da Hey, we're working on a project that needs to parse DAR files and process LF ASTs. This project is built using Stack and we're referencing the Daml repo and some of its subdirs in `extra-deps`. As long as the cabal files are there and they work I'm happy.
On a side note I think merging this PR would be beneficial for the time being even you plan to automate cabal generation as it also contains other improvements (fixing broken symlinks and a circular dependency) and the cabal files presently on `main` are not correct. Until a generative solution would be implemented for them, this PR would fix those also.

**Review by dylant-da — COMMENTED** [review]
Looks great! We're working on getting https://github.com/digital-asset/daml/pull/16755 through, after which I can generate a patch for this PR, so that all of the cabal files are tracked.

**Review by dylant-da — APPROVED** [review]
As discussed on the Slack, great work, thank you for getting this through! Having this working `generate_and_track_cabal` utility is gonna be pretty useful moving forward.

**Review by dylant-da — APPROVED** [review]

**Pull request merged**